### PR TITLE
Webhooks with OLM deployment

### PR DESCRIPTION
**Pull request by abays**
Admission webhook support with proper deployment and cleanup via OLM

### CODE REVIEW

File: bundle.Dockerfile [hunk 1]
```diff
@@ -5,7 +5,7 @@ LABEL operators.operatorframework.io.bundle.manifests.v1=manifests/
 LABEL operators.operatorframework.io.bundle.metadata.v1=metadata/
 LABEL operators.operatorframework.io.bundle.package.v1=osp-director-operator
 LABEL operators.operatorframework.io.bundle.channels.v1=alpha
-LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.2.0
+LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.1.0
```
Comment (dprince) [inline bundle.Dockerfile:8]: Remove this?

File: config/manifests/bases/osp-director-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -4,9 +4,9 @@ metadata:
   annotations:
     alm-examples: '[]'
     capabilities: Basic Install
-    operators.operatorframework.io/builder: operator-sdk-v1.2.0
-    operators.operatorframework.io/project_layout: go.kubebuilder.io/v2
+    operators.operatorframework.io/builder: operator-sdk-v1.1.0
```
Comment (dprince) [inline config/manifests/bases/osp-director-operator.clusterserviceversion.yaml:7]: I don't think there should be changes to this file either right?

File: config/webhook/kustomization.yaml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 resources:
 - manifests.yaml
-- service.yaml
+#- service.yaml
```
Comment (dprince) [inline config/webhook/kustomization.yaml:3]: why is this commented out?

File: scripts/build_and_push_images.sh [hunk 4]
```diff
@@ -4,7 +4,7 @@ set -ex
 #
 # Builds and pushes operator, bundle and index images for a given version of an operator
 #
-# NOTE: Requires opm and operator-sdk to be installed!
+# NOTE: Requires yq, make, podman, opm and operator-sdk to be installed!
```
Comment (dprince) [inline scripts/build_and_push_images.sh:7]: we don't require 'yq' again do we (I don't have it!)

File: config/webhook/kustomization.yaml (hunk 3 above)
Comment (abays) [reply config/webhook/kustomization.yaml:3]: It's not used, so I commented it out.  OLM generates the service at deploy-time automagically

File: config/manifests/bases/osp-director-operator.clusterserviceversion.yaml (hunk 2 above)
Comment (abays) [reply config/manifests/bases/osp-director-operator.clusterserviceversion.yaml:7]: Ack

File: scripts/build_and_push_images.sh (hunk 4 above)
Comment (abays) [reply scripts/build_and_push_images.sh:7]: Ack

File: bundle.Dockerfile (hunk 1 above)
Comment (abays) [reply bundle.Dockerfile:8]: Ack

**Pull request merged**